.github/workflows/release.yml

```diff
@@ -0,0 +1,64 @@
+name: Release
+
+permissions:
+  contents: write
+
+on:
+  push:
+    tags:
+      - mountpoint-s3-[0-9]+.*
+      - mountpoint-s3-[a-z]+-[0-9]+.*
+
+jobs:
+  create-release:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: taiki-e/create-gh-release-action@v1
+      with:
+        prefix: mountpoint-s3(-[a-z]+)?
+        draft: true
+        # TODO: set it true after we have changlog template reviewed.
+        changelog: false
+        # TODO: set it false after we have changlog template reviewed.
+        allow-missing-changelog: true
+        branch: main
+        title: "Mountpoint for Amazon S3 v$version"
+        token: ${{ secrets.GITHUB_TOKEN }}
+
+  upload-assets:
+    name: Build and release on target ${{ matrix.runner.target}}
+    runs-on: ${{ matrix.runner.tags }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        runner:
+        - tags: [ubuntu-20.04] # GitHub-hosted
+          target: x86_64-unknown-linux-gnu
+        - tags: [self-hosted, linux, arm64] 
```

Comment:
we need a different tag for release build hosts, which we want to be running AL2 even if we update benchmarking et al to AL2023 later.